oat-sa/tao-mobile-app
# Feature/tao 6219 sync test takers

This story is the first one in the _synchronization serie_ : the synchronization of test takers.

## How to test the story

On the endpoint 
 - open any webbrowser on the endpoint : http://spielplatz.taocloud.org/app-central-sync
 - login the backoffice (ask me an account)
 - create a Sync Manager user
 - check or add some test takers
 
On the app (or using the dev server `npm run dev` -> `http://localhost:3000`) : 
 - login with the created Sync Manager
 - Start the synchronization
 - All test takers should be included
 - Logout and try to login with a test taker

Go back to the endpoint  (could be done in mulitple times ) : 
 - remove some test takers
 - update some test takers 
 - add new test takers 

On the app ; 
 - login with the  Sync Manager
 - Start the synchronization
 - Check if you can still login with the removed account, the new accounts, etc.

## PR details
 - Some refactoring especially on the `request` side to inherit the API config
 - Add some new `app/core` components
 - Even if the task didn't require it explicitly, I've created a "temporary" layout and the controllers in order to test it (the UX will be adapted in a next story)
 - I've also added a "wipeout" feature to clear the app data
